### DISCUSSION TITLE
[@types/google.maps] | missing property google.maps.DirectionsStatus in google.maps.DirectionsResult

### DISCUSSION THREAD

**Discussion by DarkNasuada** [DefinitelyTyped/DefinitelyTyped#68919] [Issues with a @types package]
Property google.maps.DirectionsStatus is missing in the type google.maps.DirectionsResult

when i console.log the result it is in the object.

using latest version of @types/google.maps

```
const result = await DirectionsService.route(request)

//result.status throws typescript error:  Property 'status' does not exist on type 'DirectionsResult'.ts(2339)
if (result.status === 'OK') {
     DirectionsRenderer.setDirections(result)
}
```
Solution:

add following line:

`status: google.maps.DirectionsStatus;`

in the exported interface DirectionsResult (Line 1033 in index.d.ts)

[a comment by an automated account is not shown]

**Comment by kwnevarez** [comment]
Hello, this is working as intended, google.maps.DirectionsResult does not have a status field by design. Specific to the way you are calling it, if the promise returned by DirectionsService.route() resolves successfully, there is no need to check a status. And if the promise is rejected an error will be thrown indicating the reason. The reasons returned will align with the DirectionsStatus enums. 

For an example, see the function calculateAndDisplayRoute() [here](https://developers.google.com/maps/documentation/javascript/examples/directions-complex)

A fair question at this point is: why have a DirectionsStatus object at all if you don't return it? Well, there is another way to call the route() method and that is with a callback argument. Ex: DirectionsService.route(request, callback). This callback is where DirectionsStatus really shines. Here is an example where the callback takes DirectionsStatus as the argument named 'status'.

```
directionsService.route(request, function(result, status) {
    if (status == 'OK') {
      directionsRenderer.setDirections(result);
    }
  });
```
See [Displaying the DirectionsResult](https://developers.google.com/maps/documentation/javascript/directions#DirectionsStatus) for more context.
